packages/dds/tree/src/simple-tree/schemaCreationUtilities.ts

```diff
@@ -61,32 +66,49 @@ export function singletonSchema<TScope extends string, TName extends string | nu
  * Converts an enum into a collection of schema which can be used in a union.
  * @remarks
  * Currently only supports `string` enums.
+ * The string value of the enum is used as the name of the schema: ensure that its stable and unique.
+ * Consider making a dedicated schema factory with a nested scope to avoid the enum members colliding with other schema.
  * @example
  * ```typescript
- * enum Mode {
+ * const schemaFactory = new SchemaFactory("com.myApp");
+ * // An enum for use in the tree. Must have string keys.
+ * export enum Mode {
  * 	a = "A",
  * 	b = "B",
  * }
- * const ModeNodes = adaptEnum(schema, Mode);
- * type ModeNodes = NodeFromSchema<(typeof ModeNodes)[keyof typeof ModeNodes]>;
- * const nodeFromString: ModeNodes = ModeNodes(Mode.a);
- * const nodeFromSchema: ModeNodes = new ModeNodes.a();
- * const nameFromNode: Mode = nodeFromSchema.value;
- * class Parent extends schemaFactory.object("Parent", {
+ * // Define the schema for each member of the enum using a nested scope to group them together.
+ * export const ModeNodes = adaptEnum(new SchemaFactory(`${schemaFactory.scope}.Mode`), Mode);
+ * // Defined the types of the nodes which correspond to this the schema.
+ * export type ModeNodes = NodeFromSchema<(typeof ModeNodes)[keyof typeof ModeNodes]>;
+ * // An example schema which has an enum as a child.
+ * export class Parent extends schemaFactory.object("Parent", {
+ * 	// typedObjectValues extracts a list of all the fields of ModeNodes, which are the schema for each enum member.
+ * 	// This means any member of the enum is allowed in this field.
  * 	mode: typedObjectValues(ModeNodes),
  * }) {}
+ *
+ * // Example constructing a tree containing an enum node from an enum value.
+ * // The syntax `new ModeNodes.a()` is also supported.
+ * export const config = new TreeConfiguration(Parent, () => ({
+ * 	mode: ModeNodes(Mode.a),
+ * }));
+ *
+ * // Example usage of enum based nodes, showing what type to use and that `.value` can be used to read out the enum value.
+ * export function getValue(node: ModeNodes): Mode {
+ * 	return node.value;
+ * }
  * ```
  * @privateRemarks
  * TODO:
- * Extends this to support numeric enums.
- * Maybe require an explicit nested scope to group them under, or at least a warning about collisions.
- * Maybe just provide `SchemaFactory.nested` to east creating nested scopes?
- * @internal
+ * Extend this to support numeric enums.
+ * Maybe provide `SchemaFactory.nested` to ease creating nested scopes?
+ * @see {@link enumFromStrings} for a similar function that works on arrays of strings instead of an enum.
+ * @beta
  */
-export function adaptEnum<TScope extends string, const TEnum extends Record<string, string>>(
-	factory: SchemaFactory<TScope>,
-	members: TEnum,
-) {
+export function adaptEnum<
+	TScope extends string,
+	const TEnum extends Record<string, string | number>,
+>(factory: SchemaFactory<TScope>, members: TEnum) {
```

Comment:
Can we add an explicit return type here? That is required by our all of our non-minimal (now deprecated) eslint configs.

Reply:
+ the other functions

Reply:
Omitting it here was intentional. Specifying the return the here is messy since it can't refer to any of the typers in the function. Additionally doing so causes an implicit conversion to this messy hard to maintain type, which I feel is both less type safe, and annoying. For reference the return type here would be 

```typescript
(<TValue extends TEnum[keyof TEnum]>(value: TValue) => TreeNode & {
    readonly value: TValue;
}) & { readonly [Property in keyof TEnum]: TreeNodeSchemaClass<ScopedSchemaName<TScope, TEnum[Property]>, NodeKind.Object, TreeNode & {
        readonly value: TEnum[Property];
    }, never, true, unknown> & (new () => TreeNode & {
        readonly value: TEnum[Property];
    }); };
```

Given that anyone using this from outside the package will see that due to using the d.ts files, and anyone working on the package can see it in the API report, d.ts files, public docs and intelisense, I just don't see much value from including it in the .ts file explicitly when the current approach lets us express that type as `typeof factoryOut & TOut;` which is way simpler.

Generally I prefer explicit return types, but when they are a huge mess like this, I think they are more harm than good. I'd rather let the compiler come up with it, then review that (ex: in this PR, where I got that from the api report)

That said, I still see value it the argument for explicit return types: if you do really want them here I can do that, but I think it makes this code less nice of an example of how you can relatively simply write utilities that combine schema as it becomes about equal parts runtime logic and types (which duplicate the same thing).